src/coordinates.h

```diff
@@ -7,6 +7,280 @@
 #include "enums.h"
 #include "game_constants.h"
 #include "point.h"
+#include "debug.h"
+
+namespace coords
+{
+
+enum class scale {
+    map_square,
+    submap,
+    overmap_terrain,
+    segment,
+    overmap,
+    vehicle
```

Comment:
vehicle scale should possibly be split into two scales: vehicle_mount (the unrotated nominal displacement from the 0,0,0 position) and vehicle_actual (the current projection of a vehicle_mount along a tileray).  Perhaps those should be special origins.

Reply:
Yeah, I haven't delved too deeply into vehicle stuff yet.  My thoughts were that the mount coords would be origin vehicle, scale vehicle, while the projected-into-world coordinates would be origin vehicle, scale map_square.  But I'm not too tied to that idea yet.

Reply:
I'm not too particular about how they're laid out, I just hate:
`void coord_translate( int dir, const point &pivot, const point &p, point &q )` because it's not immediately clear that p and q are actually supposed to be separate types and whether pivot is the same or a different type.

If that becomes
`void coord_translate( int dir, const veh_veh_point &pivot, const veh_veh_point &p, veh_ms_point &q )` then I'm satisfied, aside from `veh_veh_point` looking kind of silly.

Reply:
`scale veh` should be `scale mnt`, so:
`void coord_translate( int dir, const veh_mnt_point &pivot, const veh_mnt_point &p, veh_ms_point &q )`
which doesn't look too bad.